tools/codegen/pkg/interfacegen/generator_test.go

```diff
@@ -15,159 +15,89 @@
 package interfacegen
 
 import (
+	"bytes"
 	"fmt"
+	"io"
 	"io/ioutil"
 	"os"
-	"os/exec"
-	"path"
-	"path/filepath"
 	"testing"
 )
 
-func TestWellKnownTemplate(t *testing.T) {
+type logFn func(string, ...interface{})
+
+// TestBazelGeneration uses the outputs files generated via the bazel rule
+// for testdata:generated_files and compares them against the golden files.
+func TestBazelGeneration(t *testing.T) {
 	tests := []struct {
-		src      string
-		baseline string
+		name, got, want string
 	}{
-		{"testdata/MetricTemplate.proto", "testdata/MetricTemplateProcessorInterface.go.baseline"},
-		{"testdata/MetricTemplate.proto", "testdata/MetricTemplateProcessorInterface.go.baseline"},
-		{"testdata/QuotaTemplate.proto", "testdata/QuotaTemplateProcessorInterface.go.baseline"},
-		{"testdata/LogTemplate.proto", "testdata/LogTemplateProcessorInterface.go.baseline"},
-		{"testdata/ListTemplate.proto", "testdata/ListTemplateProcessorInterface.go.baseline"},
-		{"testdata/NestedMessage.proto", "testdata/NestedMessageProcessor.go.baseline"},
-		{"testdata/ErrorTemplate.proto", "testdata/ErrorTemplate.baseline"},
+		{"Metrics", "testdata/metric_template_library_processor.gen.go", "testdata/MetricTemplateProcessorInterface.golden.go"},
+		{"Quota", "testdata/quota_template_library_processor.gen.go", "testdata/QuotaTemplateProcessorInterface.golden.go"},
+		{"Logs", "testdata/log_template_library_processor.gen.go", "testdata/LogTemplateProcessorInterface.golden.go"},
+		{"Lists", "testdata/list_template_library_processor.gen.go", "testdata/ListTemplateProcessorInterface.golden.go"},
+		{"Nested Message", "testdata/nested_message_library_processor.gen.go", "testdata/NestedMessageProcessorInterface.golden.go"},
```

Comment:
bazel rule for the test defines data deps, which is how these files are generated.

I have not tried to ensure that any of this works from gogland. Nor have I tried to trigger bazel build rules from with go code. I'm not sure we want to execute bazel build rules from the go code (we would have this weird bazel -> go -> bazel setup which might make it hard to access generated artifacts, etc.).

A different approach would be to have a separate test with known file descriptor protos as inputs, requiring only a call to generate.  That would not exercise the generate.bzl rules, but we could add separate tests for that.

Thoughts?